vmpooler_client/command_parser.py

```diff
@@ -65,28 +65,67 @@ def __init__(self, argv):
     self._commands = {}
     self._sub_commands = {}
 
-  def add_command(self, cmd_name, desc):
+  def add_command(self, cmd_name, desc, func=None):
     """Create a top-level command.
 
     Args:
       cmd_name |str| = The name of the command.
       desc |str| = A description of the command.
+      func |func| = A function to use for commands that do not allow sub-commands.
+        If any value is supplied it is assumed that the command will never use sub-commands.
 
     Returns:
       None
 
     Raises:
-      |KeyError| Command name has already been defined.
+      |KeyError| = Command name has already been defined.
     """
 
     if cmd_name not in self._commands:
       command = self._sub_parsers.add_parser(cmd_name, description=desc)
 
       # Add a subparser for sub-commands under a top-level command. (i.e. "list" for "vm")
-      self._commands[cmd_name] = command.add_subparsers()
+      if func is None:
```

Comment:
Not sure I care, but maybe it'd be cleaner to just do `if func` and have the positive case first, or maybe `if not func`